lib/common/templates/sass/map-deep.template

```diff
@@ -0,0 +1,80 @@
+<%
+//
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+%>
+<%
+    // for backward compatibility we need to have the user explicitly hide it
+    var showFileHeader = (this.options && this.options.hasOwnProperty('showFileHeader')) ? this.options.showFileHeader : true;
+    if(showFileHeader) {
+        var header = '';
+        header += "/*\n  Do not edit directly";
+        header += "\n  Generated on " + new Date().toUTCString();
+        header += "\n*/\n";
+        print(header);
+    }
+
+    print("\n");
+
+    // output the list of tokens as Sass variables
+    //
+    _.each(allProperties, function(prop) {
```

Comment:
Should we do this for map-flat as well?

Reply:
I've thought about it, but at that point I don't see a real use of the flat Sass map: what would be the difference between:
```
background-color: $token-some-color;
```
and
```
background-color: map-get($tokens, $some-color);
```
the flat map is essentially a "list of variables" inside a map, at that point just use a list of variables. But I understand that in this way they are more similar, flat and deep, but one possible difference is that in the case of the "flat map" if I expose the variables, they are potentially overwritable, while in the current implementation they are "safe".
I don't have a strong opinion on this, I'll let you decide (it takes me 5 mins to update the format).

Reply:
You are right, if someone wants just the variables, we have a template for that. Let's leave this as-is